### DISCUSSION TITLE
rustic.toml: path specific forget policy

### DISCUSSION THREAD

**Discussion by dimejo** [rustic-rs/rustic#460] [Q&A] [answered]
I have previously been using autorestic which allows to [configure different forget policies per location](https://autorestic.vercel.app/location/forget). Is this  possible with rustic?

**Comment by aawsome** [comment, marked as answer]
This is currently only possible via multiple config files, i.e. put your different forget options into files like `forget1.toml`, `forget2.toml`, etc and then run `rustic -P forget1 forget` and so on...

**Comment by manfredlotz** [comment]
I am also interested to have this possibility. The suggestion doesn't seem to work for me or I have not understood the suggestion.

Assume I have a test.toml with

```
[[backup.sources]]
source = "/home/manfred/repos/local"

[[backup.sources]]
source = "/home/manfred/repos/gitlab"
```

If I have a test.local.toml with

```
[[backup.sources]]
source = "/home/manfred/repos/local"

[forget]
keep-hourly = 20
keep-daily = 7
keep-weekly = 4
keep-monthly = 1
keep-yearly = 0
```

then when running `rustic -P test.local -n forget`  I get shown  snapshots from both paths would be removed.

**Reply by manfredlotz** [reply]
Oops, sorry. I overlooked the `filter...` options. So things should work with two config files

**Comment by aawsome** [comment]
A side note: With 0.5.3 you can use other config profies within a config profile file.
So, in your case you could e.g. define a `repo.toml` with all your standard configuration and then do
```
[global]
use-profile = ["repo"]

[forget]
# put your forget-profile-specific options here...
```
to define various forget profiles, e.g. `forget1.toml`, `forget2.toml`,...

Note that there is `rustic show-config` which allows you to see what resulting config rustic uses, if you have complex nested config files..

**Comment by manfredlotz** [comment]
What I have done now is to have a global config  x.toml with

``` 
[global]
log-level = "warn"
log-file = "/home/manfred/logfiles/x.log"

# repository options: These options define which backend to use and which password to use.
[repository]
repository = "/data/x"
password = "mypassword"


[backup]
exclude-if-present = [".nobackup", "CACHEDIR.TAG", "pCloudDrive/", ".pcloud/", ".snapshots/"]
one-file-system = true
```

Then two more specific ones like for example `x.home.toml`

```
[global]
use-profile = ["x"]

[[backup.sources]]
source = "/home/manfred"


[forget]
filter-paths = ["/home/manfred"]
keep-hourly = 20
keep-daily = 7
keep-weekly = 4
keep-monthly = 1
keep-yearly = 0
```

Works fine. I only have to experiment with the `keep-...` values in a way I like

**Reply by aawsome** [reply]
Great to hear that this works for you!
You could even specify the `filter-paths` under `[snapshot-filter]` - it will be also used by `forget` (unless you overwrite it in the `[forget]` section) and you have the additional benefit that other commands also use this filter, e.g. you can use `rustic -P x.home snapshots`.

**Reply by manfredlotz** [reply]
Thanks a lot. For backups I currently specify both `x.home` and `x.other`. I will take a look at your suggestion
